src/App.js

```diff
@@ -1,7 +1,201 @@
+import { Component } from "react";
 import "./App.css";
+import productData from "./data/productData";
+import formatPrice from "./helpers/formatPrice";
+
+
+class App extends Component {
+  constructor(){
+    super()
+    this.state = {
+      productsList: productData,
+      cartProductsList: [],
+      subtotal: 0,
+      tax: 0,
+      total: 0,
+      firstName: "",
+      lastName: "",
+      email: "",
+      creditCard: "",
+      zipCode: "",
+
+
+    }
+  }
+
+  handleAddToCartButton= (event) => {
+    const {productsList} = this.state
+    const {value} = event.target
+    let product = productsList.find(product =>{
+      return product.name.includes(value) 
+    })
+    
+    this.setState({
+      cartProductsList: [...this.state.cartProductsList, product],
+      subtotal: this.state.subtotal + product.price,
+      tax: this.state.subtotal * .05,
+      total: this.state.subtotal + this.state.tax
+    })
+    
+
+  }
+
+  handleBuyNowButton= () => {
+
+  };
+
+  handleFirstNameInput = (e) => {
+    this.setState({
+      firstName: e.target.value
+    })
+  };
+
+  handleLastNameInput = (e) => {
+    this.setState({
+      lastName: e.target.value
+    })
+  };
+
+  handleEmailInput = (e) => {
+    this.setState({
+      email: e.target.value
+    })
+  };
+
+  handleCreditCardInput = (e) => {
+    this.setState({
+      creditCard: e.target.value
+    })
+  };
+
+  handleZipCodeInput = (e) => {
+    this.setState({
+      zipCode: e.target.value
+    })
+  };
+  
+
+  handleFormSubmit= (e) => {
+    e.preventDefault();
+   
+  };
+
+  
+
+  render(){
+    
+    let productsListArr = this.state.productsList.map((product)=>{
+      let {name, price, img, description} = product;
+      return(
+        <div>
+          <h3>{name}</h3>
+          <div>Price: {formatPrice(price)}</div>
+          <button className="products-button" type="submit" onClick={this.handleAddToCartButton}>Add To Cart</button>
+          <img src={img} alt={name}/>
+          <div>{description}</div>
+        </div>
+      )
+    })
+
+    // let cartProductsListArr = this.state.cartProductsList.map((product)=>{
+    //   return(
+    //       <li>
+    //         {this.state.product.name}: {this.state.product.price}
+    //       </li>
+    //   )
+    // })
```

Comment:
```suggestion
  let cartProductsListArr = this.state.cartProductsList.map((product)=>{
   return(
        <li>
          {this.state.product.name}: {this.state.product.price}
        </li>
      )
     })
```
This would have worked perfectly if your `handleAddToCartButton` method was working properly. You were really close!